src/rsz/src/Resizer.cc

```diff
@@ -3648,4 +3648,79 @@ void Resizer::resetInputSlews()
   input_slew_map_.clear();
 }
 
+void Resizer::eliminateDeadLogic()
+{
+  std::vector<const Instance*> queue;
+  std::set<const Instance*> kept_instances;
+
+  auto keepInst = [&](const Instance* inst) {
+    if (!kept_instances.count(inst)) {
+      kept_instances.insert(inst);
+      queue.push_back(inst);
+    }
+  };
+
+  auto keepPinDriver = [&](Pin* pin) {
+    PinSet* drivers = network_->drivers(pin);
+    if (drivers) {
+      for (const Pin* drvr_pin : *drivers) {
+        if (auto inst = network_->instance(drvr_pin)) {
+          keepInst(inst);
+        }
+      }
+    }
+  };
+
+  auto top_inst = network_->topInstance();
+  if (top_inst) {
+    auto iter = network_->pinIterator(top_inst);
+    while (iter->hasNext()) {
+      Pin* pin = iter->next();
+      Net* net = network_->net(network_->term(pin));
+      PinSet* drivers = network_->drivers(net);
+      if (drivers) {
+        for (const Pin* drvr_pin : *drivers) {
+          if (auto inst = network_->instance(drvr_pin)) {
+            keepInst(inst);
+          }
+        }
+      }
+    }
+    delete iter;
+  }
+
+  for (auto inst : network_->leafInstances()) {
+    if (!isLogicStdCell(inst) || dontTouch(inst)) {
+      keepInst(inst);
+    } else {
+      auto iter = network_->pinIterator(inst);
+      while (iter->hasNext()) {
+        Pin* pin = iter->next();
+        Net* net;
+        if ((net = network_->net(pin)) && dontTouch(net)) {
+          keepInst(inst);
+        }
+      }
+      delete iter;
+    }
+  }
+
+  for (int i = 0; i < queue.size(); i++) {
+    auto iter = network_->pinIterator(queue[i]);
+    while (iter->hasNext()) {
+      keepPinDriver(iter->next());
+    }
+    delete iter;
+  }
+
+  int remove_count = 0;
+  for (auto inst : network_->leafInstances()) {
+    if (!kept_instances.count(inst)) {
+      sta_->deleteInstance((Instance*) inst);
+      remove_count++;
+    }
+  }
+  logger_->report("Removed {} instances", remove_count);
```

Comment:
You may have floating nets left behind that should also be cleaned up.

Reply:
Leaving a load with no driver is an error.  Under what circumstance would that happen?

Reply:
What I am saying is: `find_floating_nets` only detects nets which have loads but no drivers, so the kind of nets this new elimination command is leaving (which have both no drivers and no loads) are not picked up by it.

Reply:
`find_floating_nets` is just for reporting and doesn't clean anything up.  I suppose it is a matter of definition but I would include a net with no connections at all as floating.  It might not strictly be a violation but it is useless and makes little sense.

Regardless I think we should remove such nets.

Reply:
> find_floating_nets is just for reporting and doesn't clean anything up.

I am not claiming otherwise.

> Regardless I think we should remove such nets.

OK

Reply:
Are you wanting to defer this cleanup to another PR?